doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+
+    printf("Default sample period: %i seconds\n", dflt->sample_period);
+
+    /* Configure data collection. */
+
+    void *log_storage_context         = NULL;
+    void *log_upload_strategy_context = NULL;
+
+    /* The internal memory log storage distributed with Kaa SDK. */
+    error = ext_unlimited_log_storage_create(&log_storage_context,
+        kaa_client_get_context(kaa_client)->logger);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Create a strategy based on timeout. */
+    error = ext_log_upload_strategy_create(
+        kaa_client_get_context(kaa_client), &log_upload_strategy_context,
+        KAA_LOG_UPLOAD_BY_TIMEOUT_STRATEGY);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Strategy will upload logs every 5 seconds. */
+    error = ext_log_upload_strategy_set_upload_timeout(log_upload_strategy_context, 5);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Specify log bucket size constraints. */
+    kaa_log_bucket_constraints_t bucket_sizes = {
+         .max_bucket_size       = 32,   /* Bucket size in bytes. */
+         .max_bucket_log_count  = 2,    /* Maximum log count in one bucket. */
+    };
+
+    /* Initialize the log storage and strategy (by default it is not set). */
+    error = kaa_logging_init(kaa_client_get_context(kaa_client)->log_collector,
+        log_storage_context, log_upload_strategy_context, &bucket_sizes);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Start Kaa SDK's main loop. example_callback will be called 1 time per second. */
+
+    error = kaa_client_start(kaa_client, example_callback, kaa_client, 1);
+
+    /* Should get here only after Kaa stop. */
+
+    kaa_client_destroy(kaa_client);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+```
+
+</div>
+
+<div id="app-java" class="tab-pane fade" markdown="1" >
+
+```java
+java java java
+```
+
+</div>
+
+</div>
+
+## Expected output
+
+If to run application, following logs will be observed in console:
+
+```
+Default sample period: 1 seconds
+Sampled temperature: 32
+Sampled temperature: 26
+Sampled temperature: 26
+Sampled temperature: 31
+Sampled temperature: 28
+Sampled temperature: 28
+```
+Note that temperature value must be sampled 1 time per second as stated in the
+configuration scheme.
+Refer to the [Troubleshooting](#troubleshooting) section if something goes wrong.
+
+## Retrieving collected data
+
+Now it is time to stored temperature from the server.
+
+1. Grab application token. It is a token that you can copy from the main window
+    of the application the Administration UI.
+
+    ![Generate SDK](attach/app_token.png)
+1. Login to your machine running Kaa server (in case you are using sandbox
+    default username and password are *kaa / kaa*).
+1. Start MongoDB shell
+
+        mongo kaa
+1. Retrieve data using application token.
+
+        db.logs_$your_application_token$.find()
```

Comment:
Please note the application token at the place it is referred first and give a link to that place here.